General fixes

Novanet- opened the pull request.

Novanet- committed:
Added a couple of extensions (don't be scared of them =P)

Novanet- committed:
Refactored StarSystem.cs
Added a type enumerator to the interface declaration to remove the need for casting upon enumaration of star system
Renaming to fit C# code style
Reorganisation and regions for readability
Created a sub method in Generate() to reduce overly-long method
Fixed namespaces for all classes in Controllers folder
Adjusted visiblity of class members to be as strict as possible
Nullability annotations to help avoid runtime NullReferences
Readonly modifiers to reduce accidental mutation

Novanet- commented:
Don't merge this until I've removed the WIP flag

Novanet- committed:
Added type parameter to IEnumerable in StarSystem.cs

Novanet- committed:
Replaced IEnumerator<T> implemntaion with a GetEnumator() that uses yield

Much less code, and more maintanable

Novanet- committed:
Adjusted Controllers namespace

Novanet- committed:
Adjusted some casting to make it safer

Novanet- committed:
UI tweaking

Novanet- committed:
Refactored CameraFollowPlayer.cs

Used attribute based component loading
Renaming for consistent code style

Novanet- committed:
Refactored RadorObject.cs

Renaming
Reduced nesting
Member visibility

Novanet- committed:
Refactored WorldX.cs

Minor formatting

Novanet- committed:
Refactored CommodityRecord.cs

Renaming
Regions

Novanet- committed:
Refactored EmptyOrbital.cs

Minor formatting

Novanet- committed:
Refactored Galaxy.cs

Renaming and other minor stuff

Novanet- committed:
Refactored IInteractable.cs

Minor, regions

Novanet- committed:
Refactored InfoInteraction.cs

Novanet- committed:
Refactored Orbital.cs and OrbitalInteraction.cs

Novanet- committed:
Refactored Planetoid.cs

Changed arrays to strongly typed collections
Used LINQ where possible
Other general refactorings

Novanet- committed:
Refactored TradeInteraction.cs

Renaming, regions, and mvoed struct to partial class

Novanet- committed:
Refactored InfoTab.cs

Novanet- committed:
Refactored InteractionTab.cs

Novanet- committed:
Resaved scene

Novanet- committed:
Refactored InteractionUI.cs

Novanet- committed:
Refactored TabButton.cs

Novanet- committed:
Refactored TradeTab.cs

Novanet- committed:
Refactored WorldUI.cs

Novanet- commented:
@pterolatypus @Arkainnihx Ok you can see if this is ready to merge now

Novanet- committed:
Additional refactoring for TradeInteraction.cs

Novanet- committed:
Added post processing stack

Novanet- committed:
Usage of Component attribute

Novanet- committed:
Stronger collection types in StarSystem.cs

Novanet- committed:
Force text serialisation mode

Novanet- committed:
Updated submodule Assets/EditorExtensions/InputBinder

Novanet- committed:
Updated submodule Assets/EditorExtensions/Unity3D-ComponentAttribute

Novanet- committed:
Updated submodule Assets/EditorExtensions/unity-themes

Novanet- committed:
Updated submodule Assets/EditorExtensions/unity-ui-extensions

Novanet- committed:
Fucking fixed shitty git shit

Novanet- committed:
Fixed component shit

Novanet- committed:
fuck this submodule

Novanet- committed:
and this one

Novanet- committed:
and also this

pterolatypus committed:
Removed Klak and KlakUI packages
Moved initial system load call to Start to avoid initialisation conflicts

pterolatypus committed:
Removed some redundant meta files

pterolatypus commented:
Apart from the changes that we need to make to StarSystem enumeration, this seems all good. I have a few convention/style points I think we need to agree on before continuing though:

- I'm not sold on the judicious use of #region. I don't think it makes the code more readable, it just adds clutter and I'm concerned it will conflict with the Clean Code style we want to migrate to.
 - Prefacing interfaces with I is a common convention, but I'm not a fan.
 - Formatting variable names differently depending on their scope etc. feels unnecessary.
 - We should probably convert as many public fields to serialized-private as possible (topic for another branch maybe)
 - Do we want a deeper code hierarchy, with more namespaces?

pterolatypus committed:
Adjusted postprocessing a bit

pterolatypus committed:
Refactored CameraFollowPlayer
Refactored to clean code style
Added following arbitrary game objects
Refactored position locking and following to separate methods

pterolatypus committed:
Refactored GameController
Refactored to clean code style
Converted some public fields to serialized private
Removed relocate method (obsolete)

pterolatypus committed:
Refactored PlayerShipController
Refactored to clean code style
Moved some public fields to private serialized
Extracted some fields to properties

pterolatypus committed:
Reverted Interactable name
